### PR TITLE
Edit update testing

### PR DESCRIPTION
**Pull request by ahzia**
In this Pull we added tests for:
- editing the task description.
- updating an item's 'completed' status.
- updating an item's index value upon drag/drop actions.
- the "Clear all completed" function.

### CODE REVIEW

File: src/test/task.test.js [hunk 1]
```diff
@@ -84,3 +85,106 @@ describe('Delete a task method', () => {
...
+    addTask(tasks, 'third task');
+    updateStorage(tasks);
+
+    output = [
+      { description: 'third task', index: 1, completed: false },
+      { description: 'The Task has been Edited', index: 2, completed: false },
+      { description: 'second task', index: 3, completed: false },
+    ];
+    // console.log('tasks ', tasks);
+
```
Comment (george-shammar) [inline src/test/task.test.js:154]: - [ ] Kindly clean up the commented `console.log`  in your codebase. It does not make your codebase look tidy.
- [ ] Please document your testing in the README by including the testing process in the README.

[OPTIONAL]

- [ ] You have way too many comments in this module. Most of them are unnecessary, they'll simply make the code untidy. It's clear what each line of code does, so adding comments at every other line may not be necessary. I advise that you remove them.

**Review by george-shammar — CHANGES_REQUESTED** [review]
Hi @benwmx and @ahzia  , 

Good job so far!

:heavy_check_mark: Your tests are passing and you have covered all the required functions. :clap: 

There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by benwmx** [comment]
thank you @george-shammar for your thorough review, I really appreciate that.

File: src/test/task.test.js (hunk 1 above)
Comment (ahzia) [reply src/test/task.test.js]: Thanks For Your Comments,
About your optional comment, In the [JS Testing Best Practises - AAA](https://github.com/goldbergyoni/javascript-testing-best-practices#-%EF%B8%8F-12-structure-tests-by-the-aaa-pattern) topic, It says that we should separate each part of AAA with comments. You can check the code examples on the topic.

**Review by oxenprogrammer — APPROVED** [review]
Hi **@ahzia**  and **@benwmx** , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**